components/lib/crash/src/test/java/mozilla/components/lib/crash/service/GleanCrashReporterServiceTest.kt

```diff
@@ -0,0 +1,237 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.lib.crash.service
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.work.testing.WorkManagerTestInitHelper
+import mozilla.components.lib.crash.Crash
+import mozilla.components.lib.crash.GleanMetrics.CrashMetrics
+import mozilla.components.service.glean.Glean
+import mozilla.components.support.test.whenever
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Assert.assertFalse
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.verify
+import java.io.File
+
+@RunWith(AndroidJUnit4::class)
+class GleanCrashReporterServiceTest {
+    private val context: Context
+        get() = ApplicationProvider.getApplicationContext()
+
+    private fun setupGleanCrashReporterServiceTest() {
+        WorkManagerTestInitHelper.initializeTestWorkManager(
+            ApplicationProvider.getApplicationContext())
+
+        Glean.enableTestingMode()
+        Glean.initialize(context)
+    }
+
+    @Test
+    fun `GleanCrashReporterService records native code crashes`() {
+        var initialValue = 0
+        setupGleanCrashReporterServiceTest()
+
+        run {
+            // Because of how Glean is implemented, it can potentially persist information between
+            // tests or even between test classes, so we compensate by capturing the initial value
+            // to compare to.
+            if (CrashMetrics.crashCount[GleanCrashReporterService.NATIVE_CODE_CRASH_KEY].testHasValue()) {
```

Comment:
nit: not super important, at your discretion here and in the other locations :)

```
initialvalue = try {
  CrashMetrics.crashCount[GleanCrashReporterService.NATIVE_CODE_CRASH_KEY].testGetValue()
} catch(..) { 0 }
```

Reply:
This is good!